pkg/operators/clustermanager/controllers/migrationcontroller/migration_controller.go

```diff
@@ -174,70 +208,129 @@ func removeStorageVersionMigrations(
 func applyStorageVersionMigrations(ctx context.Context, migrationClient migrationv1alpha1client.StorageVersionMigrationsGetter, recorder events.Recorder) error {
 	errs := []error{}
 	for _, file := range migrationRequestFiles {
-		_, _, err := applyStorageVersionMigration(
-			migrationClient,
+		required, err := parseStorageVersionMigrationFile(
 			func(name string) ([]byte, error) {
 				template, err := manifests.ClusterManagerManifestFiles.ReadFile(name)
 				if err != nil {
 					return nil, err
 				}
 				return assets.MustCreateAssetFromTemplate(name, template, struct{}{}).Data, nil
 			},
-			recorder,
 			file)
 		if err != nil {
 			errs = append(errs, err)
+			continue
+		}
+
+		_, _, err = applyStorageVersionMigration(migrationClient, required, recorder)
+		if err != nil {
+			errs = append(errs, err)
+			continue
 		}
 	}
 
 	return operatorhelpers.NewMultiLineAggregate(errs)
 }
 
+// syncStorageVersionMigrationsCondition sync the migration condition based on all the StorageVersionMigrations status
+// 1. migrationSucceeded is true only when all the StorageVersionMigrations resources succeed.
+// 2. migrationSucceeded is false when any of the StorageVersionMigrations resources failed or running
+func syncStorageVersionMigrationsCondition(ctx context.Context, migrationClient migrationv1alpha1client.StorageVersionMigrationsGetter) (metav1.Condition, error) {
+	for _, file := range migrationRequestFiles {
+		required, err := parseStorageVersionMigrationFile(
+			func(name string) ([]byte, error) {
+				template, err := manifests.ClusterManagerManifestFiles.ReadFile(name)
+				if err != nil {
+					return nil, err
+				}
+				return assets.MustCreateAssetFromTemplate(name, template, struct{}{}).Data, nil
+			},
+			file)
+		if err != nil {
+			return metav1.Condition{}, err
```

Comment:
Same as above question, does this mean if no required migration file, there will be no condition? and the crd_status_controller will keep requeueing every 5 seconds?

Reply:
Also, if not supported, will the condition always be false and crd_status_controller will keep requeueing every 5 seconds?